earn/src/components/borrow/modal/tab/AddUniswapNFTAsCollateralTab.tsx

```diff
@@ -0,0 +1,381 @@
+import {
+  ALOE_II_UNISWAP_NFT_MANAGER_ADDRESS,
+  UNISWAP_NONFUNGIBLE_POSITION_MANAGER_ADDRESS,
+} from '../../../../data/constants/Addresses';
+import { MarginAccount } from '../../../../data/MarginAccount';
+import { getValueOfLiquidity, tickToPrice, UniswapNFTPosition } from '../../../../data/Uniswap';
+import { truncateDecimals } from '../../../../util/Numbers';
+import TokenPairIcons from '../../../common/TokenPairIcons';
+
+const SECONDARY_COLOR = '#CCDFED';
+const TERTIARY_COLOR = '#4b6980';
+const ITEMS_PER_PAGE = 2;
+
+const GAS_ESTIMATE_WIGGLE_ROOM = 110; // 10% wiggle room
+
+const UniswapNFTPositionsPage = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 8px;
+  min-height: 280px;
+`;
+
+enum ConfirmButtonState {
+  APPROVE_NFT_MANAGER,
+  APPROVING,
+  PENDING,
+  LOADING,
+  READY,
+}
+
+function getConfirmButton(state: ConfirmButtonState): { text: string; enabled: boolean } {
+  switch (state) {
+    case ConfirmButtonState.APPROVE_NFT_MANAGER:
+      return { text: 'Approve', enabled: true };
+    case ConfirmButtonState.APPROVING:
+      return { text: 'Approving', enabled: false };
+    case ConfirmButtonState.PENDING:
+      return { text: 'Pending', enabled: false };
+    case ConfirmButtonState.LOADING:
+      return { text: 'Loading', enabled: false };
+    case ConfirmButtonState.READY:
+      return { text: 'Confirm', enabled: true };
+    default:
+      return { text: 'Confirm', enabled: false };
+  }
+}
+
+export const UniswapNFTPositionButtonWrapper = styled.button.attrs((props: { active: boolean }) => props)`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  justify-content: center;
+  padding: 8px 16px;
+  gap: 8px;
+  border-radius: 8px;
+  opacity: ${(props) => (props.active ? 1 : 0.25)};
+  filter: ${(props) => (props.active ? 'none' : 'grayscale(100%)')};
+  cursor: pointer;
+  background-color: rgba(26, 41, 52, 1);
+
+  &:hover {
+    filter: none;
+    opacity: 1;
+  }
+`;
+
+type UniswapNFTPositionButtonProps = {
+  marginAccount: MarginAccount;
+  uniswapNFTPosition: UniswapNFTPosition;
+  isActive: boolean;
+  onClick: () => void;
+};
+
+function UniswapNFTPositionButton(props: UniswapNFTPositionButtonProps) {
+  const { marginAccount, uniswapNFTPosition, isActive, onClick } = props;
+
+  const { token0, token1 } = uniswapNFTPosition;
+
+  const [minPrice, maxPrice] = useMemo(() => {
+    const min = tickToPrice(
+      uniswapNFTPosition.tickLower,
+      uniswapNFTPosition.token0.decimals,
+      uniswapNFTPosition.token1.decimals,
+      true
+    );
+    const max = tickToPrice(
+      uniswapNFTPosition.tickUpper,
+      uniswapNFTPosition.token0.decimals,
+      uniswapNFTPosition.token1.decimals,
+      true
+    );
+    return [min, max];
+  }, [
+    uniswapNFTPosition.tickLower,
+    uniswapNFTPosition.tickUpper,
+    uniswapNFTPosition.token0.decimals,
+    uniswapNFTPosition.token1.decimals,
+  ]);
```

Comment:
I don't think `useMemo` is really necessary here, and code will read better without it.